File: sdk/c#/PicarioXPO.RenderAPI.V2/OverlayKeyExtractors/LocationOverlayKey.cs
using System.Collections.Generic;
using System.Linq;

namespace PicarioXPO.RenderAPI.V2.OverlayKeyExtractors
{
    internal class LocationOverlayKey : OverlayKey
    {
        public override string GetValues(IEnumerable<XpoUrlOverlay> overlays)
        {
            var xpoUrlOverlays = overlays as XpoUrlOverlay[] ?? overlays.ToArray();
            var max = xpoUrlOverlays.GetMaxOverlayNumber();

            for (var i = 0; i <= max; i++)
            {
                var urlObject = xpoUrlOverlays.LastOrDefault(x => x.Index == i);
                if (urlObject != null)
                    AddToList(urlObject.OverlayLocation);
                else if (i != max)
                    AddEmpty();
            }

            if (IsEmpty()) return "";

            return "p.ol=" + GetUrlValue();
        }
    }
}

File: sdk/js/PicarioXPO.RenderAPI.V2/XpoUrlGenerator/LocationOverlayKey.ts
﻿class LocationOverlayKey extends OverlayKey {

    getValues(overlays: Array<XpoUrlOverlay>) {
        var max = UrlGeneratorModule.getMaxOverlayNumber(overlays);

        for (var i = 0; i <= max; i++) {
            var index = overlays.map(e => e.getIndex()).lastIndexOf(i);

            var urlOverlay = overlays[index];
            if (urlOverlay != null)
                this.addToList(urlOverlay.getOverlayLocation());
            else if (i != max)
                this.addEmpty();
        }

        if (this.isEmpty()) return "";

        return "p.ol=" + this.getUrlValue();
    }
}